src/repr/type_intf.ml

```diff
@@ -273,10 +284,28 @@ module type DSL = sig
   (** The type for staged operations. *)
 
   val stage : 'a -> 'a staged
-  (** [stage x] stages [x]. *)
+  (** [stage x] stages [x] where [x] would typically be a partially applied
+      function. *)
```

Comment:
```suggestion
  (** [stage x] stages [x], where [x] would typically be a function that is
      expensive to construct. *)
```

It's not quite right to say that the generic functions are partially-applied, since they (ideally) perform as much computation as possible after each argument is passed.

Reply:
Got it 👍